index.js

```diff
@@ -1,285 +0,0 @@
-     }else{ 
-        for (let k = 0; k < row; k++) { 
-            let tempObj = {ground: gold,x:k*50 ,y:i*50};
-            mapArr[i].push(tempObj);
-            
-        }
-        }
-    }
-  
-    return mapArr;
-  }
-// makeMap/ mapArr to be changed for  generateMap/ matrix at line 56, , 166, 180, 219,220  
-
- 
-function draw(){ 
-    background(220);
-
-   if (showStart){
-    image(startScreen,0,0,0,1000);
-   }
-     else if(gameRunning && goldCount==0 ){
-    textSize(32);
-    text(`Fuel: ${fuel}`,500,35);
-    if(ironCount<5){
-        text(`Iron Collected: ${ironCount}`,50,35);
-    }else {
-        text(`Ironhack activated!`,50,35);
-    }
-    fill(51);
-    
- for(let i=0;i<mapArr.length;i++){
-        let tempArr = mapArr[i];
-        for (let j=0; j<tempArr.length;j++){
-             image(tempArr[j].ground,tempArr[j].x,tempArr[j].y);
-        }
-    }
-
- image (playerNorm,player.x,player.y);
-
-   if (fuel < 0 && !goldCount>=1){
-        resetButton = false;
-        gameRunning = false;
-    }
-    //out of fuel
-    }else if(showStart==false &&gameRunning== false && goldCount==0){
-    image(gameOverScreen,0,0,0,1000);
-    const button = document.querySelector('#restart');
-     button.style.cssText= `visibility: visible`;
-    }
-    // win screen
-    else if(goldCount>=1){
-        resetButton= false;
-        gameRunning= false;
-        image(winScreen,0,0,0,1000);
-        const button = document.querySelector('#restart');
-        button.style.cssText= `visibility: visible`;
-    }
-}
-function startGame (){
-    resetButton=true;
-    gameRunning=true;
-    showStart = false;
-    
-    document.querySelector('#start').style.cssText = `visibility: hidden`;
-   //generateMap();
-    makeMap();
-
-}
-function resetGame(){
-    resetButton=true;
-    gameRunning=true;
-    player.x = 300;
-    player.y = 0;
-    fuel = 15;
-    goldCount = 0;
-    ironCount =0;
-    
-    document.querySelector('#restart').style.cssText = `visibility: hidden`;
-    //generateMap();
-    makeMap();
-
-}
-
-function randomMaterial(){
-    let rand= randomNum(100);
-    if (rand>=9){
-        return dirt;
-    }else if(rand<=4){
-        return iron;
-    }else{
-        return oil;
-    }
-}
-
-function randomNum(max){
- return Math.floor(Math.random()* max);
-}
-
-
-function keyPressed(){
-    if(keyCode === LEFT_ARROW){
-        player.x===0 ? null :player.x-= 50;
-        fuel= fuel - 1;
-       // digSound.play();
-    }else if ( keyCode === RIGHT_ARROW){
-        player.x===600 ? null : player.x+= 50;
-        fuel= fuel - 1;
-      //  digSound.play();
-    }  else if (keyCode === DOWN_ARROW){
-        fuel= fuel - 1;
-     player.y+= 50;
-    // digSound.play();
-    }else if (keyCode === UP_ARROW){
-        player.y===0 ? null :player.y-= 50;
-        fuel= fuel - 1;
-    
-    }
-    
-
-    for(let i=0;i<mapArr.length;i++){
-        let tempArr = mapArr[i];
-        for (let j=0; j<tempArr.length;j++){
```

Comment:
these if/else statements can be re-written as a function that can be reused

something that looks something like this:

```javascript
function updateWhenHittingResource(resource, update) {
  if(player.x===tempArr[j].x && player.y===tempArr[j].y && tempArr[j].ground=== resource) {
       update()         
  }
}

function digRock() {
  if(ironCount < 5) {
    player.y-= 50
    console.log("cant mine rock yet");
  } else {
      tempArr[j].ground = dug;
    }  
}
updateWhenHittingResource(iron, digRock)
```

Reply:
this is still obvs. somewhat pseudo code-ish but the general idea should be clear. 
have one function for the logic of "if player hits the resource" 
then supply that function with the function "what happens if player hits that resource"